applicationinsights/channel/AsynchronousSender.py

```diff
@@ -123,6 +129,14 @@ def _run(self):
                 # if we didn't get any items from the queue, we're done here
                 if len(data) == 0:
                     break
+                
+                # Drop extra events from the queue until it no larger than send_buffer_size.
+                # This is a necessary evil to prevent the backlog from bloating forever in bandwidth-limited environments
+                eventsToDrop = max(0, local_queue._queue.qsize() - self._send_buffer_size)
```

Comment:
I was under the impression that that was the point of separating _max_queue_length (500) and _send_buffer_size (100). If the send buffer gets backed up it will still be able to buffer up to 500 before it overflows